Pipeline InitContainer goes into CrashLoopBackoff

**What happened?**:

Encountering `CrashLoopBackOff` on the init container for all pipeline Pods (using Pachyderm 1.11.0 in managed Kubernetes environment). On this line: https://github.com/pachyderm/pachyderm/blob/4c5e1ab70e4b372697d9ebefff391b41b02e6807/etc/worker/init.go#L35

This was when the pipeline was already running for ~5days and no changes were done to the pipeline spec or Pod spec directly.

Not sure why the init container was restarted (unsure if Pachyderm might trigger an initContainer restart itself or if it was something externally?). Both the user and storage container have status Running (for about ~5d), and Pod is also still up - it seems to be just the init container.

Logs of the init container:

```
panic: open /pach-bin/worker: text file busy
29/12/2020 09:22:24 
29/12/2020 09:22:24 goroutine 1 [running]:
29/12/2020 09:22:24 main.main()
29/12/2020 09:22:24  /app/init.go:36 +0x78
```

`kubectl describe` of one of the pipelines (only initContainer is crashing...):
``` 
❯ kubectl describe po pipeline-v1-dqrvs -n production
Name:         pipeline-v1-dqrvs
Namespace:    production
Priority:     0
Node:         lnx4837/10.75.238.5
Start Time:   Wed, 23 Dec 2020 15:16:25 +0100
Labels:       app=pipeline-v1
              component=worker
              pipelineName=REMOVED
              suite=pachyderm
Annotations:  authTokenHash: REMOVED
              cni.projectcalico.org/podIP: 10.42.4.110/32
              cni.projectcalico.org/podIPs: 10.42.4.110/32
              pipelineName: REMOVED
              specCommit: 64df5fa218334260a9d6dd16fd7111bd
              version: 1.11.0
Status:       Running
IP:           10.42.4.110
IPs:
  IP:           10.42.4.110
Controlled By:  ReplicationController/pipeline-v1
Init Containers:
  init:
    Container ID:  docker://d28aa30ea083839063cd28b6bbe16642bde1f42517d2f875ee37caa6ede7a624
    Image:         pachyderm/worker:1.11.0
    Image ID:      docker-pullable://pachyderm/worker@sha256:80c665308461125fe158fc6f2f51d6487e91fea2b7abaa39f278e1590b33266e
    Port:          <none>
    Host Port:     <none>
    Command:
      /app/init
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Tue, 29 Dec 2020 12:49:06 +0100
      Finished:     Tue, 29 Dec 2020 12:49:06 +0100
    Ready:          False
    Restart Count:  62
    Requests:
      cpu:        2
      memory:     2G
    Environment:  <none>
    Mounts:
      /pach-bin from pach-bin (rw)
      /pfs from pachyderm-worker (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from pachyderm-worker-token-cw6nh (ro)
Containers:
  user:
    Container ID:  docker://66f50e3e6dceb00ba164c73392a5e0bcb350467c1391284655ed89f5a30e5a25
    Image:         REMOVED
    Image ID:      REMOVED
    Port:          <none>
    Host Port:     <none>
    Command:
      /pach-bin/worker
    State:          Running
      Started:      Wed, 23 Dec 2020 15:17:01 +0100
    Ready:          True
    Restart Count:  0
    Requests:
      cpu:     2
      memory:  2G
    Environment:
      PPS_PIPELINE_NAME:     REMOVED
      PACH_ROOT:             /pach
      PACH_NAMESPACE:        production
      PPS_WORKER_IP:          (v1:status.podIP)
      PPS_ETCD_PREFIX:       pachyderm/1.7.0/pachyderm_pps
      PPS_POD_NAME:          pipeline-v1-dqrvs (v1:metadata.name)
      PPS_SPEC_COMMIT:       64df5fa218334260a9d6dd16fd7111bd
      PPS_WORKER_GRPC_PORT:  80
      PEER_PORT:             653
      STORAGE_BACKEND:       AMAZON
      GOOGLE_BUCKET:         <set to the key 'google-bucket' in secret 'pachyderm-storage-secret'>        Optional: true
      GOOGLE_CRED:           <set to the key 'google-cred' in secret 'pachyderm-storage-secret'>          Optional: true
      MICROSOFT_CONTAINER:   <set to the key 'microsoft-container' in secret 'pachyderm-storage-secret'>  Optional: true
      MICROSOFT_ID:          <set to the key 'microsoft-id' in secret 'pachyderm-storage-secret'>         Optional: true
      MICROSOFT_SECRET:      <set to the key 'microsoft-secret' in secret 'pachyderm-storage-secret'>     Optional: true
      MINIO_BUCKET:          <set to the key 'minio-bucket' in secret 'pachyderm-storage-secret'>         Optional: true
      MINIO_ENDPOINT:        <set to the key 'minio-endpoint' in secret 'pachyderm-storage-secret'>       Optional: true
      MINIO_ID:              <set to the key 'minio-id' in secret 'pachyderm-storage-secret'>             Optional: true
      MINIO_SECRET:          <set to the key 'minio-secret' in secret 'pachyderm-storage-secret'>         Optional: true
      MINIO_SECURE:          <set to the key 'minio-secure' in secret 'pachyderm-storage-secret'>         Optional: true
      MINIO_SIGNATURE:       <set to the key 'minio-signature' in secret 'pachyderm-storage-secret'>      Optional: true
      AMAZON_REGION:         <set to the key 'amazon-region' in secret 'pachyderm-storage-secret'>        Optional: true
      AMAZON_BUCKET:         <set to the key 'amazon-bucket' in secret 'pachyderm-storage-secret'>        Optional: true
      AMAZON_ID:             <set to the key 'amazon-id' in secret 'pachyderm-storage-secret'>            Optional: true
      AMAZON_SECRET:         <set to the key 'amazon-secret' in secret 'pachyderm-storage-secret'>        Optional: true
      AMAZON_TOKEN:          <set to the key 'amazon-token' in secret 'pachyderm-storage-secret'>         Optional: true
      AMAZON_VAULT_ADDR:     <set to the key 'amazon-vault-addr' in secret 'pachyderm-storage-secret'>    Optional: true
      AMAZON_VAULT_ROLE:     <set to the key 'amazon-vault-role' in secret 'pachyderm-storage-secret'>    Optional: true
      AMAZON_VAULT_TOKEN:    <set to the key 'amazon-vault-token' in secret 'pachyderm-storage-secret'>   Optional: true
      AMAZON_DISTRIBUTION:   <set to the key 'amazon-distribution' in secret 'pachyderm-storage-secret'>  Optional: true
      CUSTOM_ENDPOINT:       <set to the key 'custom-endpoint' in secret 'pachyderm-storage-secret'>      Optional: true
      RETRIES:               <set to the key 'retries' in secret 'pachyderm-storage-secret'>              Optional: true
      TIMEOUT:               <set to the key 'timeout' in secret 'pachyderm-storage-secret'>              Optional: true
      UPLOAD_ACL:            <set to the key 'upload-acl' in secret 'pachyderm-storage-secret'>           Optional: true
      REVERSE:               <set to the key 'reverse' in secret 'pachyderm-storage-secret'>              Optional: true
      PART_SIZE:             <set to the key 'part-size' in secret 'pachyderm-storage-secret'>            Optional: true
      MAX_UPLOAD_PARTS:      <set to the key 'max-upload-parts' in secret 'pachyderm-storage-secret'>     Optional: true
      DISABLE_SSL:           <set to the key 'disable-ssl' in secret 'pachyderm-storage-secret'>          Optional: true
      NO_VERIFY_SSL:         <set to the key 'no-verify-ssl' in secret 'pachyderm-storage-secret'>        Optional: true
    Mounts:
      /pach from pach-dir-volume (rw)
      /pach-bin from pach-bin (rw)
      /pachyderm-storage-secret from pachyderm-storage-secret (rw)
      /pfs from pachyderm-worker (rw)
      /var/run/docker.sock from docker (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from pachyderm-worker-token-cw6nh (ro)
  storage:
    Container ID:  docker://4b691abd5e419efcf2ac31ceaf1edb544c55038eee70c5cdc152d458fafaa98f
    Image:         pachyderm/pachd:1.11.0
    Image ID:      docker-pullable://pachyderm/pachd@sha256:568c20b30f30865527bb15e7436b48680192f1a7c9b33e716326808b8f900266
    Port:          <none>
    Host Port:     <none>
    Command:
      /pachd
      --mode
      sidecar
    State:          Running
      Started:      Wed, 23 Dec 2020 15:17:01 +0100
    Ready:          True
    Restart Count:  0
    Requests:
      cpu:     0
      memory:  64M
    Environment:
      PACH_ROOT:                         /pach
      PACH_NAMESPACE:                    production
      BLOCK_CACHE_BYTES:                 64M
      PFS_CACHE_SIZE:                    16
      STORAGE_BACKEND:                   AMAZON
      PPS_WOKER_GRPC_PORT:               80
      PORT:                              650
      HTTP_PORT:                         652
      PEER_PORT:                         653
      PPS_SPEC_COMMIT:                   64df5fa218334260a9d6dd16fd7111bd
      GC_PERCENT:                        50
      STORAGE_BACKEND:                   AMAZON
      GOOGLE_BUCKET:                     <set to the key 'google-bucket' in secret 'pachyderm-storage-secret'>        Optional: true
      GOOGLE_CRED:                       <set to the key 'google-cred' in secret 'pachyderm-storage-secret'>          Optional: true
      MICROSOFT_CONTAINER:               <set to the key 'microsoft-container' in secret 'pachyderm-storage-secret'>  Optional: true
      MICROSOFT_ID:                      <set to the key 'microsoft-id' in secret 'pachyderm-storage-secret'>         Optional: true
      MICROSOFT_SECRET:                  <set to the key 'microsoft-secret' in secret 'pachyderm-storage-secret'>     Optional: true
      MINIO_BUCKET:                      <set to the key 'minio-bucket' in secret 'pachyderm-storage-secret'>         Optional: true
      MINIO_ENDPOINT:                    <set to the key 'minio-endpoint' in secret 'pachyderm-storage-secret'>       Optional: true
      MINIO_ID:                          <set to the key 'minio-id' in secret 'pachyderm-storage-secret'>             Optional: true
      MINIO_SECRET:                      <set to the key 'minio-secret' in secret 'pachyderm-storage-secret'>         Optional: true
      MINIO_SECURE:                      <set to the key 'minio-secure' in secret 'pachyderm-storage-secret'>         Optional: true
      MINIO_SIGNATURE:                   <set to the key 'minio-signature' in secret 'pachyderm-storage-secret'>      Optional: true
      AMAZON_REGION:                     <set to the key 'amazon-region' in secret 'pachyderm-storage-secret'>        Optional: true
      AMAZON_BUCKET:                     <set to the key 'amazon-bucket' in secret 'pachyderm-storage-secret'>        Optional: true
      AMAZON_ID:                         <set to the key 'amazon-id' in secret 'pachyderm-storage-secret'>            Optional: true
      AMAZON_SECRET:                     <set to the key 'amazon-secret' in secret 'pachyderm-storage-secret'>        Optional: true
      AMAZON_TOKEN:                      <set to the key 'amazon-token' in secret 'pachyderm-storage-secret'>         Optional: true
      AMAZON_VAULT_ADDR:                 <set to the key 'amazon-vault-addr' in secret 'pachyderm-storage-secret'>    Optional: true
      AMAZON_VAULT_ROLE:                 <set to the key 'amazon-vault-role' in secret 'pachyderm-storage-secret'>    Optional: true
      AMAZON_VAULT_TOKEN:                <set to the key 'amazon-vault-token' in secret 'pachyderm-storage-secret'>   Optional: true
      AMAZON_DISTRIBUTION:               <set to the key 'amazon-distribution' in secret 'pachyderm-storage-secret'>  Optional: true
      CUSTOM_ENDPOINT:                   <set to the key 'custom-endpoint' in secret 'pachyderm-storage-secret'>      Optional: true
      RETRIES:                           <set to the key 'retries' in secret 'pachyderm-storage-secret'>              Optional: true
      TIMEOUT:                           <set to the key 'timeout' in secret 'pachyderm-storage-secret'>              Optional: true
      UPLOAD_ACL:                        <set to the key 'upload-acl' in secret 'pachyderm-storage-secret'>           Optional: true
      REVERSE:                           <set to the key 'reverse' in secret 'pachyderm-storage-secret'>              Optional: true
      PART_SIZE:                         <set to the key 'part-size' in secret 'pachyderm-storage-secret'>            Optional: true
      MAX_UPLOAD_PARTS:                  <set to the key 'max-upload-parts' in secret 'pachyderm-storage-secret'>     Optional: true
      DISABLE_SSL:                       <set to the key 'disable-ssl' in secret 'pachyderm-storage-secret'>          Optional: true
      NO_VERIFY_SSL:                     <set to the key 'no-verify-ssl' in secret 'pachyderm-storage-secret'>        Optional: true
      STORAGE_UPLOAD_CONCURRENCY_LIMIT:  100
    Mounts:
      /pach from pach-dir-volume (rw)
      /pachyderm-storage-secret from pachyderm-storage-secret (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from pachyderm-worker-token-cw6nh (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  pach-bin:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  pachyderm-worker:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  pach-dir-volume:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  pachyderm-storage-secret:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  pachyderm-storage-secret
    Optional:    false
  docker:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/docker.sock
    HostPathType:
  pachyderm-worker-token-cw6nh:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  pachyderm-worker-token-cw6nh
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason   Age                       From              Message
  ----     ------   ----                      ----              -------
  Warning  BackOff  2m50s (x1338 over 4h52m)  kubelet, lnx4837  Back-off restarting failed container
```

**What you expected to happen?**:
Init container not to go into `CrashLoopBackOff`. They should be robust to restarts following: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#detailed-behavior.

**How to reproduce it (as minimally and precisely as possible)?**:
Unsure, because due to managed Kubernetes environment I cant determine what exactly caused it. Perhaps manually restarting the initContainer?

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`): 1.19.4
- Pachyderm CLI and pachd server version (use `pachctl version`): pachctl 1.11.7 & pachd 1.11.0
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): Rancher via 3rd party hosting
- OS (e.g. from /etc/os-release): Oracle Linux Server 7.9
- Others:

--- a/etc/worker/init.go
+++ b/etc/worker/init.go
@@ -6,6 +6,17 @@ import (
 )
 
 func cp(src, dst string) error {
+	// init containers can be restarted spuriously, so make sure we actually need to copy
+	info, err := os.Stat(dst)
+	if err == nil {
+		// if the file already exists and is executable, assume it's correct
+		if info.Mode() == os.ModePerm {
+			return nil
+		}
+	} else if !os.IsNotExist(err) {
+		return err
+	}
+
 	in, err := os.Open(src)
 	if err != nil {
 		return err

